src/plugins/workspace/public/plugin.ts

```diff
@@ -106,9 +139,28 @@ export class WorkspacesPlugin implements Plugin<{}, {}, WorkspacesPluginSetupDep
     }
   }
 
+  private getFilterLinks(
+    workspace: WorkspaceAttribute | null | undefined,
+    allNavLinks: ChromeNavLink[]
+  ) {
+    if (!workspace) return allNavLinks;
+    const features = workspace.features ?? [];
+    return allNavLinks.filter((item) => features.includes(item.id));
+  }
+
   public start(core: CoreStart) {
     // If workspace feature is disabled, it will not load the workspace plugin
     if (core.uiSettings.get('workspace:enabled') === false) {
+      // set default value for filtered nav links
```

Comment:
Can we move this block to a private method with a meaningful name like `registerFilterLinksWhenFlagOff` so that others will know what is doing for this block from the method name?